### DISCUSSION TITLE
Call from invalid thread error while creating brush in Task.

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10713] [Q&A] [answered]
I want to create several SolidColorBrush in Task for adding resource.

Here is my code:
`
Task.Run(() => {
                var k = new SolidColorBrush(Colors.Black);
            });
`

After the program ran, it reports an error: Call from invalid thread

In WPF it should run well. I am not sure but if I can only create SolidColorBrush in UITherad by Avalonia?

**Comment by maxkatz6** [comment, marked as answer]
Yes, you can create mutable components only on UI thread. It's expected. 

But you can create ImmutableSolidColorBrush from other threads.

**Reply by kyuranger** [reply]
> Yes, you can create mutable components only on UI thread. It's expected.
> 
> But you can create ImmutableSolidColorBrush from other threads.

Thanks a lot.
